ui/src/app/workflows/components/workflow-dag/workflow-dag.tsx

```diff
@@ -185,7 +189,11 @@ export class WorkflowDag extends React.Component<WorkflowDagProps, WorkflowDagRe
             return <Loading />;
         }
         const {nodes, edges} = this.prepareGraph();
-        this.layoutGraph(nodes, edges);
+        if (this.state.renderingMode === 'Fast' || (this.state.renderingMode === 'Auto' && nodes.length > 200)) {
```

Comment:
Need to empirically test at how many nodes to switch to fast renderer instead of nice